hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java

```diff
@@ -913,6 +938,10 @@ private void cleanServerCache(ServerName server, Throwable regionException) {
     if (ClientExceptionsUtil.isMetaClearingException(regionException)) {
       // We want to make sure to clear the cache in case there were location-related exceptions.
       // We don't to clear the cache for every possible exception that comes through, however.
+      MetricsConnection metrics = asyncProcess.connection.getConnectionMetrics();
```

Comment:
i think we should do this as a separate jira

Reply:
That's cool - figured since it was tiny it could go together. But given the conversation of this fix only being required in certain versions makes sense to split them out.

Reply:
yea, since it's just a metrics change it should hopefully be simple to put up separately. in general the hbase folks prefer "one change per jira". if we were adding new metrics specifically about our change here, that would make sense. but in this case you're sort of fixing another separate but related bug

part of the reason is for ease of managing history, and another is it makes it easier for release managers to curate changelogs that encompass all of the changes done.